### ISSUE TITLE
Completions fails in 0.12.0

### ISSUE THREAD

**Issue by DonJayamanne** [davidhalter/jedi#1113]
* Windows 10
* Default system python is Anaconda 3.6.4 (this is what is in current path)
* Jedi version 0.12.0

Code:
```python
import sys
import jedi
src = """import sys
sys.
"""
script = jedi.Script(src, 1, len('src.'), 'one.py', sys_path=sys.path)
print(len(script.completions()))
```

This throws the following error:

<section>
<details>
<pre>
Traceback (most recent call last):
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\evaluate\compiled\subprocess\__init__.py", line 202, in _send
    is_exception, traceback, result = pickle_load(self._process.stdout)
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\_compatibility.py", line 462, in pickle_load
    return pickle.loads(data, encoding='bytes')
EOFError: Ran out of input

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\api\__init__.py", line 174, in completions
    completions = completion.completions()
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\api\completion.py", line 97, in
completions
    completion_names = self._get_context_completions()
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\api\completion.py", line 172, in _get_context_completions
    completion_names = list(self._get_keyword_completion_names(allowed_keywords))
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\api\completion.py", line 209, in _get_keyword_completion_names
    yield keywords.KeywordName(self._evaluator, k)
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\api\keywords.py", line 28, in __init__
    self.parent_context = evaluator.builtins_module
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\evaluate\cache.py", line 40, in
wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\evaluate\__init__.py", line 115, in builtins_module
    return compiled.get_special_object(self, u'BUILTINS')
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\evaluate\compiled\__init__.py",
line 27, in get_special_object
    evaluator.compiled_subprocess.get_special_object(identifier)
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\evaluate\compiled\subprocess\__init__.py", line 93, in wrapper
    kwargs=kwargs,
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\evaluate\compiled\subprocess\__init__.py", line 158, in run
    return self._send(id(evaluator), function, args, kwargs)
  File "C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\pythonFiles\jedi\evaluate\compiled\subprocess\__init__.py", line 205, in _send
    raise InternalError("The subprocess %s has crashed." % self._executable)
jedi.api.exceptions.InternalError: The subprocess C:\msys64\usr\bin\python3.6.exe has crashed.
</pre>
</details>
</section>

**Comment by DonJayamanne** [comment]
Looks like jedi is using some other interpreter in a subprocess. I believe its using one that gets returned by `jedi.get_default_environment()`.

* The default environment on my system is `Anaconda 3.6.4`
* None of the Jedi Api returns the above environment
* I'm unable to use the above Anaconda environment in the jedi api (see #1114)

**Comment by DonJayamanne** [comment]
`C:\msys64\usr\bin\python3.6.exe`

**Comment by blueyed** [comment]
@DonJayamanne 
Can you try https://github.com/davidhalter/jedi/pull/1124 to see why it crashes?

**Comment by blueyed** [comment]
@DonJayamanne 
Gentle ping to try https://github.com/davidhalter/jedi/pull/1124 - i.e. check out the PR in `pythonx/jedi`.

**Comment by davidhalter** [comment]
I have merged #1176. So many things have changed, please run the script with latest master and report back if there's still an error (I don't have an anaconda environment where I could reproduce it).

**Comment by davidhalter** [comment]
Closing, because it's probably fixed. If it's not, please say that and I'll reopen.
